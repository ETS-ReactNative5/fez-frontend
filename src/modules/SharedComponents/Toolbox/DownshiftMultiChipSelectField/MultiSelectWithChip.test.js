import React from 'react';
import {
    renderInput,
    renderSuggestion,
    renderSuggestions,
    getSuggestions,
    MultiSelectWithChip,
} from './MultiSelectWithChip';


const setup = (testProps, isShallow = true) => {
    const props = {
        classes: {},
        optionsList: [],
        ...testProps
    }
    return getElement(MultiSelectWithChip, props, isShallow);
};

describe('MultiSelectWithChip component', () => {
    it('should render properly', () => {
        const testProps = {
            classes: {
                container: 'test1',
                clearAll: 'test2'
            },
            getInputProps: jest.fn(),
            getItemProps: jest.fn({}),
            handleClearAll: jest.fn(),
            handleDelete: jest.fn(),
            handleInputChange: jest.fn(),
            handleKeyDown: jest.fn(),
            highlightedIndex: 0,
            inputValue: 'test3',
            isOpen: true,
            label: 'test4',
            optionsList: [],
            placeholder: 'test3',
            selectedItems: ['item1'],
        };
        const wrapper1 = setup(testProps);
        expect(toJson(wrapper1)).toMatchSnapshot();

        testProps.isOpen = false;
        const wrapper2 = setup(testProps);
        expect(toJson(wrapper2)).toMatchSnapshot();

        delete testProps.selectedItems;
        const wrapper3 = setup(testProps);
        expect(toJson(wrapper3)).toMatchSnapshot();
    });
});

describe('renderInput helper', () => {
    it('should return html as expected', () => {
        const test = renderInput({
            InputProps: {
                testProp: 'test1'
            },
            classes: {
                inputRoot: 'test2',
                inputInput: 'test3'
            },
            ref: 'test4',
            testAttr: 'test5'
        });

        expect(test).toMatchSnapshot();
    });
});

describe('renderSuggestion helper', () => {
    it('should return html as expected', () => {
        const testProps = {
            suggestion: {
                label: 'test1'
            },
            index: 1,
            itemProps: {
                prop1: 'test2',
                prop2: 'test3'
            },
            highlightedIndex: 1,
            selectedItems: false
        };
        const test1 = renderSuggestion(testProps);
        expect(test1).toMatchSnapshot();

        const test2 = renderSuggestion({
            ...testProps,
            selectedItems: 'This is test1'
        });
        expect(test2).toMatchSnapshot();
    });
});

describe('renderSuggestions helper', () => {
    it('should render properly', () => {
        const inputValue = 'test';
        const optionsList = [
            { label: 'test1' },
            { label: 'test2' }
        ];
        const getItemProps = jest.fn();
        const highlightedIndex = 1;
        const selectedItems = [];
        const render = shallow(
            <div>
                {renderSuggestions({
                    inputValue,
                    optionsList,
                    getItemProps,
                    highlightedIndex,
                    selectedItems
                })}
            </div>
        );
        expect(toJson(render)).toMatchSnapshot();
        expect(getItemProps).toBeCalledTimes(optionsList.length);
    });

});

describe('getSuggestions helper', () => {
    it('should return html as expected', () => {
        const value = 'test';
        const suggestions = [
            { label: 'test1' },
            { label: 'test2' },
            { label: 'test3' },
            { label: 'test4' },
            { label: 'test5' },
            { label: 'test6' }
        ];
        const test1 = getSuggestions(value, suggestions);
        expect(test1).toMatchSnapshot();

        const test2 = getSuggestions('', suggestions);
        expect(test2).toMatchSnapshot();
    });
});
