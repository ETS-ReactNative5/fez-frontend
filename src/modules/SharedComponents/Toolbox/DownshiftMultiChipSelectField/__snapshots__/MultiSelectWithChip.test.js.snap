// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectWithChip component should render properly 1`] = `
<div
  className="test1"
>
  <WithStyles(Grid)
    alignContent="flex-end"
    alignItems="flex-end"
    container={true}
  >
    <WithStyles(Grid)
      item={true}
      xs={true}
    >
      <TextField
        InputProps={
          Object {
            "classes": Object {
              "input": undefined,
              "root": undefined,
            },
            "inputRef": undefined,
          }
        }
        fullWidth={true}
        label="test4"
        required={false}
        select={false}
        variant="standard"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs="auto"
    >
      <WithStyles(Tooltip)
        title="Clear all selections"
      >
        <WithStyles(IconButton)
          className="test2"
          color="secondary"
          component="span"
          onClick={[MockFunction]}
        >
          <pure(CloseIcon) />
        </WithStyles(IconButton)>
      </WithStyles(Tooltip)>
    </WithStyles(Grid)>
    <WithStyles(Paper)
      square={true}
    />
  </WithStyles(Grid)>
</div>
`;

exports[`MultiSelectWithChip component should render properly 2`] = `
<div
  className="test1"
>
  <WithStyles(Grid)
    alignContent="flex-end"
    alignItems="flex-end"
    container={true}
  >
    <WithStyles(Grid)
      item={true}
      xs={true}
    >
      <TextField
        InputProps={
          Object {
            "classes": Object {
              "input": undefined,
              "root": undefined,
            },
            "inputRef": undefined,
          }
        }
        fullWidth={true}
        label="test4"
        required={false}
        select={false}
        variant="standard"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs="auto"
    >
      <WithStyles(Tooltip)
        title="Clear all selections"
      >
        <WithStyles(IconButton)
          className="test2"
          color="secondary"
          component="span"
          onClick={[MockFunction]}
        >
          <pure(CloseIcon) />
        </WithStyles(IconButton)>
      </WithStyles(Tooltip)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;

exports[`MultiSelectWithChip component should render properly 3`] = `
<div
  className="test1"
>
  <WithStyles(Grid)
    alignContent="flex-end"
    alignItems="flex-end"
    container={true}
  >
    <WithStyles(Grid)
      item={true}
      xs={true}
    >
      <TextField
        InputProps={
          Object {
            "classes": Object {
              "input": undefined,
              "root": undefined,
            },
            "inputRef": undefined,
          }
        }
        fullWidth={true}
        label="test4"
        required={false}
        select={false}
        variant="standard"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;

exports[`getSuggestions helper should return html as expected 1`] = `
Array [
  Object {
    "label": "test1",
  },
  Object {
    "label": "test2",
  },
  Object {
    "label": "test3",
  },
  Object {
    "label": "test4",
  },
  Object {
    "label": "test5",
  },
]
`;

exports[`getSuggestions helper should return html as expected 2`] = `Array []`;

exports[`renderInput helper should return html as expected 1`] = `
<TextField
  InputProps={
    Object {
      "classes": Object {
        "input": "test3",
        "root": "test2",
      },
      "inputRef": "test4",
      "testProp": "test1",
    }
  }
  required={false}
  select={false}
  testAttr="test5"
  variant="standard"
/>
`;

exports[`renderSuggestion helper should return html as expected 1`] = `
<WithStyles(MenuItem)
  component="div"
  prop1="test2"
  prop2="test3"
  selected={true}
  style={
    Object {
      "fontWeight": 400,
    }
  }
>
  test1
</WithStyles(MenuItem)>
`;

exports[`renderSuggestion helper should return html as expected 2`] = `
<WithStyles(MenuItem)
  component="div"
  prop1="test2"
  prop2="test3"
  selected={true}
  style={
    Object {
      "fontWeight": 500,
    }
  }
>
  test1
</WithStyles(MenuItem)>
`;

exports[`renderSuggestions helper should render properly 1`] = `
<div>
  <WithStyles(MenuItem)
    component="div"
    key="test1"
    selected={false}
    style={
      Object {
        "fontWeight": 400,
      }
    }
  >
    test1
  </WithStyles(MenuItem)>
  <WithStyles(MenuItem)
    component="div"
    key="test2"
    selected={true}
    style={
      Object {
        "fontWeight": 400,
      }
    }
  >
    test2
  </WithStyles(MenuItem)>
</div>
`;
